runtime/src/lib.rs

```diff
@@ -119,3 +123,132 @@ impl ffi::IRDescriptorTableEntry {
         metadata
     }
 }
+
+#[doc(alias = "IRRuntimeDrawPrimitives")]
+pub fn draw_primitives(
+    encoder: &ProtocolObject<dyn MTLRenderCommandEncoder>,
+    primitive_type: MTLPrimitiveType,
+    vertex_start: usize,
+    vertex_count: usize,
+    instance_count: usize,
+    base_instance: usize,
+) {
+    let mut dp = ffi::IRRuntimeDrawParams {
+        u_1: ffi::IRRuntimeDrawParams_u {
+            draw: ffi::IRRuntimeDrawArgument {
+                vertexCountPerInstance: vertex_count as u32,
+                instanceCount: instance_count as u32,
+                startVertexLocation: vertex_start as u32,
+                startInstanceLocation: base_instance as u32,
+            },
+        },
+    };
+    unsafe {
+        encoder.setVertexBytes_length_atIndex(
+            NonNull::new(&raw mut dp).unwrap().cast(),
+            size_of_val(&dp),
+            ffi::kIRArgumentBufferDrawArgumentsBindPoint as usize,
+        );
+        let mut non_indexed_draw = ffi::kIRNonIndexedDraw;
+        encoder.setVertexBytes_length_atIndex(
+            NonNull::new(&raw mut non_indexed_draw).unwrap().cast(),
+            size_of_val(&non_indexed_draw),
+            ffi::kIRArgumentBufferUniformsBindPoint as usize,
+        );
+        encoder.drawPrimitives_vertexStart_vertexCount_instanceCount_baseInstance(
+            primitive_type,
+            vertex_start,
+            vertex_count,
+            instance_count,
+            base_instance,
+        );
+    }
+}
+
+#[doc(alias = "IRMetalIndexToIRIndex")]
+pub fn metal_index_to_ir_index(index_type: MTLIndexType) -> u16 {
+    index_type.0 as u16 + 1
```

Comment:
Is a potential overflow relevant here? The original implementation may rely on C++ just wrapping back to zero, but here we'll just panic (in debug)

Reply:
Since [`MTLIndexType` is an `enum` with currently only two constant values `0` and `1` defined](https://docs.rs/objc2-metal/0.3.1/src/objc2_metal/generated/MTLStageInputOutputDescriptor.rs.html#139-144), I think we're safe to not worry about this and rely on a debug panic if there's ever a variant of value `u16::MAX`.